test/alternator/test_returnvalues.py

```diff
@@ -83,6 +83,28 @@ def test_put_item_returnvalues(test_table_s):
     with pytest.raises(ClientError, match='ValidationException'):
         test_table_s.put_item(Item={'p': p, 'a': 'hello'}, ReturnValues='none')
 
+# Testing feature which returns values only on delete condition failures.
+def test_put_item_returnvalues_on_condition_check_failure(test_table_s):
+    p = random_string()
+    test_table_s.put_item(Item={'p': p, 's': 'dog'})
+    # Succesful conditional put doesn't return values.
+    ret=test_table_s.put_item(Item={'p': p, 's': 'cat'},
+        ReturnValuesOnConditionCheckFailure='ALL_OLD',
+        ConditionExpression='s = :v1',
+        ExpressionAttributeValues={':v1' : 'dog'})
+    assert not 'Attributes' in ret
+    # Failed conditional delete returns old values.
```

Comment:
The following is conditional put, not conditional delete

Reply:
thx